maro/simulator/scenarios/vm_scheduling/business_engine.py

```diff
@@ -113,9 +123,25 @@ def _load_configs(self):
         self._max_cpu_oversubscription_rate: float = self._config.MAX_CPU_OVERSUBSCRIPTION_RATE
         self._max_memory_oversubscription_rate: float = self._config.MAX_MEM_OVERSUBSCRIPTION_RATE
         self._max_utilization_rate: float = self._config.MAX_UTILIZATION_RATE
+
         # Load PM related configs.
-        self._pm_amount: int = self._cal_pm_amount()
-        self._kill_all_vms_if_overload = self._config.KILL_ALL_VMS_IF_OVERLOAD
+        self._region_amount: int = sum(
+            len(item) for item in self._find_item(key="region", dictionary=self._config.architecture)
+        )
+        self._zone_amount: int = sum(
+            len(item) for item in self._find_item(key="zone", dictionary=self._config.architecture)
+        )
+        self._data_center_amount: int = sum(
+            len(item) for item in self._find_item(key="data_center", dictionary=self._config.architecture)
+        )
+
+        self._cluster_amount: int = len(self._config.components.cluster)
```

Comment:
**BUG:** here the `self._cluster/rack/pm_amount` is only the number of cluster/rack/pm types, not the total amount.

Reply:
`self._config.components.cluster`'s length is the total amount of cluster.
`pm_amount` is counted by 
```python
sum(amount for amount in self._find_item(key="amount", dictionary=self._config.components))
```